server/__init__.py

```diff
@@ -102,6 +106,33 @@ def onDescribeResource(event):
     describeResponse['basePath'] = '/api/v1'
 
 
+def onJobSave(event):
+    # Patch a bug with how girder_worker's Girder task spec's 'api_url' is set
+    # with Vagrant port forwarding and Nginx proxies
+    # This is fundamentally a problem with "rest.getApiUrl"
+    job = event.info
+    if job['handler'] == 'worker_handler':
+        # We need to do this for all job statuses, since due to the way that
+        # Job.save is overridden, the local model may be desynchronized from
+        # the database after saving; this is fine, since girder_worker
+        # (where it matters) will always load directly from the correct entry
+        # in the database
+        def replaceHost(url):
+            # TODO: get this from the server config or the request
+            return 'http://127.0.0.1:8080' + url[url.find('/api/v1'):]
+
+        job['kwargs'] = json_util.loads(job['kwargs'])
+        for specValue in itertools.chain(
```

Comment:
A small example of the expected structure of a job could make this code faster to understand.